### PR TITLE
Add owner to share

### PR DESCRIPTION
**Pull request by klaustopher**
Adds a virtual share for the owner so that they are included in the list of shares

---

Implements https://community.openproject.org/projects/openproject/work_packages/56243/activity

### CODE REVIEW

File: app/controllers/shares_controller.rb [hunk 1]
```diff
@@ -44,17 +43,19 @@ class SharesController < ApplicationController
...
 
-    render Shares::ModalBodyComponent.new(strategy: sharing_strategy, shares: @shares, errors: @errors), layout: nil
+    render Shares::ModalBodyComponent.new(strategy: sharing_strategy, errors: @errors), layout: nil
   end
 
   def create # rubocop:disable Metrics/AbcSize,Metrics/PerceivedComplexity
     overall_result = []
     @errors = ActiveModel::Errors.new(self)
 
+    had_shares_before_adding = sharing_strategy.shares.present?
```
Comment (aaron-contreras) [inline app/controllers/shares_controller.rb]: Thank you.

File: app/controllers/shares_controller.rb [hunk 2]
```diff
@@ -311,31 +314,21 @@ def load_entity # rubocop:disable Metrics/AbcSize
...
   def load_selected_shares
     @selected_shares = Member.includes(:principal)
                              .of_entity(@entity)
                              .where(id: params[:share_ids])
   end
+
+  def query_params
+    params
+      .slice(:filters, :sortBy, :groupBy)
+      .permit! # ParamsToQueryService will parse the data, so we can permit everything here
```
Comment (aaron-contreras) [inline app/controllers/shares_controller.rb:332]: TIL this exists

File: app/models/sharing_strategies/base_strategy.rb [hunk 3]
```diff
@@ -94,9 +95,14 @@ def shares_query(params) # rubocop:disable Metrics/AbcSize
...
 
-      @query.order(name: :asc) unless params[:sortBy]
+      @query.order(name: :asc) unless @query_params[:sortBy]
 
       @query
     end
+
+    def shares(reload: false)
+      @shares = nil if reload
+      @shares ||= query.results
```
Comment (aaron-contreras) [inline app/models/sharing_strategies/base_strategy.rb:105]: Nice trick

**Review by aaron-contreras — APPROVED** [review]
This might be just a detail and not very important. I understand now that the owner is a virtual share amongst the other list of shares.

It is however either being included with every role filter unless there are now "real" shares matching the filter in which case it's removed. I leave a demo below:

https://github.com/opf/openproject/assets/61627014/667a5590-1dcb-4256-a9f5-214c17e2a774

Other than this, everything else is working as it used to 🍹

**Comment by klaustopher** [comment]
Oh yes, this makes sense ... I think we can add a bit more smartness and look at the filter and the roles in there. Will add this tomorrow morning. Thanks for the review

**Comment by klaustopher** [comment]
I added some code so that the owner is just shown when filtering for users and when the role matches

**Review by aaron-contreras — APPROVED** [review]
A+ work!

🍹

**Pull request merged**